scipy/sparse/_dok.py

```diff
@@ -140,27 +140,34 @@ def get(self, key, default=0.0):
             key = key[0]
         return self._dict.get(key, default)
 
-    # override IndexMixin.__getitem__ for 1d case until fully implemented
-    def __getitem__(self, key):
-        if self.ndim == 2:
-            return super().__getitem__(key)
-
-        if isinstance(key, tuple) and len(key) == 1:
-            key = key[0]
-        INT_TYPES = (int, np.integer)
-        if isinstance(key, INT_TYPES):
-            if key < 0:
-                key += self.shape[-1]
-            if key < 0 or key >= self.shape[-1]:
-                raise IndexError('index value out of bounds')
-            return self._get_int(key)
-        else:
-            raise IndexError('array/slice index for 1d dok_array not yet supported')
-
     # 1D get methods
     def _get_int(self, idx):
         return self._dict.get(idx, self.dtype.type(0))
 
+    def _get_slice(self, idx):
+        i_range = range(*idx.indices(self.shape[0]))
+        return self._get_array(list(i_range))
+
+    def _get_array(self, idx):
+        idx = np.asarray(idx)
```

Comment:
This call can blow up the range, is that something we are worried about?

Reply:
This is how the method `_asindices` handles the input `idx`. I can guess that we aren't worried about it blowing up because we don't support sparse non-boolean arrays as indices, and we handle boolean arrays (in `_validate_indices`) by converting them using `ix.nonzero()`.